---
title: "Effect of phenotype"
author: "Paula Ruiz Rodriguez"
date: today
date-format: "DD/MM/YYYY"
format:
  html: 
    code-fold: true
    page-layout: article
    embed-resources: true
    smooth-scroll: true
    theme: minty
    toc: true
    toc-depth: 5
    toc-location: left
    toc-title: Summary
    number-sections: true
    grid:
      sidebar-width: 200px
      body-width: 1200px
      margin-width: 100px
      gutter-width: 1.5rem  
runtime: shiny
editor: visual
---
```{r libraries, message=FALSE, warning=FALSE, echo=FALSE, results = FALSE}
# List of CRAN and Bioconductor packages to check and install if necessary
packages <- c("dplyr", "ggplot2", "ggthemes", "writexl", "readxl", "DT", "plotly", "devtools","gplots","pheatmap","ggplotify","VennDiagram")

bioconductor_packages <- c("biomaRt", "clusterProfiler", "DESeq2", "org.Hs.eg.db", 
                           "org.Bt.eg.db", "aPEAR", "DOSE", "enrichplot", "HTSFilter", "DESeq2", "rrvgo")

# Function to check and install CRAN packages
check_and_install <- function(package) {
  if (!require(package, character.only = TRUE)) {
    install.packages(package, dependencies = TRUE)
    library(package, character.only = TRUE)
  }
}

# Function to check and install Bioconductor packages
check_and_install_bioc <- function(package) {
  if (!require(package, character.only = TRUE)) {
    BiocManager::install(package)
    library(package, character.only = TRUE)
  }
}

# Apply the function to each CRAN package
sapply(packages, check_and_install)

# Apply the function to each Bioconductor package
if (!require("BiocManager", character.only = TRUE)) {
  install.packages("BiocManager")
}
sapply(bioconductor_packages, check_and_install_bioc)

# Install ggvenn from GitHub using devtools
if (!require("ggvenn", character.only = TRUE)) {
  devtools::install_github("yanlinlin82/ggvenn")
  library("ggvenn", character.only = TRUE)
}
```

```{r colors, echo = FALSE}
color_virulence <- c("High" = "#FF0000", "Medium" = "#FFA500", "Low" = "#00BFFF","Control" = "lightgrey")
colores_bact <- c("L5"="#995200","L6"="#02a834","A1"="#d1ae00","A4"="#ff9cdb", "Control"="grey")
```

```{r metadataTable, echo=FALSE}
dataTHP1 <- data.frame(
  sample_name = c("R0366", "R0368", "R0370", "R0201", "R0378", "R0380", "R0420", "R0422", "R0424", "R0426", "R0428", "R0430", "R0168", "R0169", "R0171", "R0187", "R0189", "R0190", "R0318", "R0320", "R0322", "R0330", "R0332", "R0334", "R0196", "R0197", "R0390", "R0394", "R0404"),
  row.names = c("R0366", "R0368", "R0370", "R0201", "R0378", "R0380", "R0420", "R0422", "R0424", "R0426", "R0428", "R0430", "R0168", "R0169", "R0171", "R0187", "R0189", "R0190", "R0318", "R0320", "R0322", "R0330", "R0332", "R0334", "R0196", "R0197", "R0390", "R0394", "R0404"),
  BACTERIA = c("A1", "A1", "A1", "A1", "A1", "A1", "A4", "A4", "A4", "A4", "A4", "A4", "Control", "Control", "Control", "Control", "Control", "Control", "L5", "L5", "L5", "L5", "L5", "L5", "L6", "L6", "L6", "L6", "L6"),
  infection_date = c(1, 1, 1, 3, 3, 3, 1, 1, 1, 3, 3, 3, 1, 1, 1, 3, 3, 3, 1, 1, 1, 3, 3, 3, 3, 3, 1, 1, 3),
  HOST = rep("THP1", 29),
  combo = c("A1-THP1", "A1-THP1", "A1-THP1", "A1-THP1", "A1-THP1", "A1-THP1", "A4-THP1", "A4-THP1", "A4-THP1", "A4-THP1", "A4-THP1", "A4-THP1", "Control-THP1", "Control-THP1", "Control-THP1", "Control-THP1", "Control-THP1", "Control-THP1", "L5-THP1", "L5-THP1", "L5-THP1", "L5-THP1", "L5-THP1", "L5-THP1", "L6-THP1", "L6-THP1", "L6-THP1", "L6-THP1", "L6-THP1"),
  VIRULENCE = c("Medium", "Medium", "Medium", "Medium", "Medium", "Medium", "Low", "Low", "Low", "Low", "Low", "Low", "Control", "Control", "Control", "Control", "Control", "Control", "High", "High", "High", "High", "High", "High", "Medium", "Medium", "Medium", "Medium", "Medium")
)
dataBo <- data.frame(
  sample_name = c("R0230", "R0231", "R0372", "R0374", "R0376", "R0386", "R0348", "R0350", "R0352", "R0360", "R0362", "R0364", "R0235", "R0236", "R0238", "R0251", "R0252", "R0253", "R0324", "R0326", "R0328", "R0336", "R0338", "R0340", "R0212", "R0213", "R0226", "R0227", "R0228", "R0408"),
  row.names = c("R0230", "R0231", "R0372", "R0374", "R0376", "R0386", "R0348", "R0350", "R0352", "R0360", "R0362", "R0364", "R0235", "R0236", "R0238", "R0251", "R0252", "R0253", "R0324", "R0326", "R0328", "R0336", "R0338", "R0340", "R0212", "R0213", "R0226", "R0227", "R0228", "R0408"),
  BACTERIA = c("A1", "A1", "A1", "A1", "A1", "A1", "A4", "A4", "A4", "A4", "A4", "A4", "Control", "Control", "Control", "Control", "Control", "Control", "L5", "L5", "L5", "L5", "L5", "L5", "L6", "L6", "L6", "L6", "L6", "L6"),
  infection_date = c(3, 3, 1, 1, 1, 3, 1, 1, 1, 3, 3, 3, 1, 1, 1, 3, 3, 3, 1, 1, 1, 3, 3, 3, 1, 1, 3, 3, 3, 1),
  HOST = rep("BoMac", 30),
  combo = c("A1-BoMac", "A1-BoMac", "A1-BoMac", "A1-BoMac", "A1-BoMac", "A1-BoMac", "A4-BoMac", "A4-BoMac", "A4-BoMac", "A4-BoMac", "A4-BoMac", "A4-BoMac", "Control-BoMac", "Control-BoMac", "Control-BoMac", "Control-BoMac", "Control-BoMac", "Control-BoMac", "L5-BoMac", "L5-BoMac", "L5-BoMac", "L5-BoMac", "L5-BoMac", "L5-BoMac", "L6-BoMac", "L6-BoMac", "L6-BoMac", "L6-BoMac", "L6-BoMac", "L6-BoMac"),
  VIRULENCE = c("Medium", "Medium", "Medium", "Medium", "Medium", "Medium", "High", "High", "High", "High", "High", "High", "Control", "Control", "Control", "Control", "Control", "Control", "Low", "Low", "Low", "Low", "Low", "Low", "Medium", "Medium", "Medium", "Medium", "Medium", "Medium")
)
combined_metadata <- rbind(dataTHP1, dataBo)
```

```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
get_one_to_one_orthologs <- function(gene_list) {
  # Usar la base de datos biomart de Ensembl para Bos taurus
  ensembl <- useMart("ensembl", dataset = "btaurus_gene_ensembl")

  # Atributos para recuperar ortÃ³logos humanos de genes de Bos taurus
  attributes <- c('ensembl_gene_id', 'hsapiens_homolog_ensembl_gene', 'hsapiens_homolog_orthology_type', 'hsapiens_homolog_associated_gene_name')

  # Recuperar ortÃ³logos para la lista proporcionada de genes de Bos taurus
  orthologs <- getBM(attributes = attributes,
                     filters = 'ensembl_gene_id',
                     values = gene_list,
                     mart = ensembl)

  # Filtrar por ortÃ³logos uno-a-uno
  orthologs_one_to_one <- orthologs[orthologs$hsapiens_homolog_orthology_type == 'ortholog_one2one', ]

  # Devolver el marco de datos filtrado
  return(orthologs_one_to_one)
}
```

```{r plotmetadata, echo=FALSE}
#| label: fig-metadata
#| fig-cap: "Distribution of samples per Infection Bacteria-Host, Day and type of Compatibility."
#| warning: false
ggplot(combined_metadata, aes(x = as.factor(infection_date), fill = VIRULENCE)) +
  geom_bar(position = position_dodge()) +
  facet_grid(HOST ~ BACTERIA, scales = "free_x") +
  labs(title = "",
       x = "Infection Day",
       y = "Count",
       fill = "Compatibility Level") +
  theme_minimal() +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1, size = 12), 
    axis.text.y = element_text(size = 12), 
    axis.title.x = element_text(size = 14), 
    axis.title.y = element_text(size = 14), 
    strip.text.x = element_text(size = 12), 
    strip.text.y = element_text(size = 12), 
    plot.title = element_text(size = 16, hjust = 0.5), 
    legend.text = element_text(size = 12), 
    legend.title = element_text(size = 14),
    legend.position = "bottom" 
  ) + 
  scale_fill_manual(values = color_virulence,
                    name = "Virulence Level")
rm(combined_metadata)
```

```{r functions1, echo=FALSE}
create_and_analyze_DESeqDataSet <- function(countData, colData, design_formula) {
  colData$infection_date <- as.factor(colData$infection_date)
  dds <- DESeq2::DESeqDataSetFromMatrix(countData = countData, 
                                        colData = colData, 
                                        design = as.formula(design_formula))
  dds <- DESeq2::DESeq(dds)
  return(dds)
}
```

## Virulence Splitted by cell type

Formula for the model: 
$$
0 + \text{VIRULENCE}_{\text{type}} * \text{INFECTION}_{\text{day}}
$$
Contrast Comparisons:
$$
\text{VIRULENCE}_{x} - \text{VIRULENCE}_{Y}
$$

```{r readcounts, echo = FALSE, message=FALSE, warning=FALSE}
count_THP1 <- read.table("humanRawCounts.txt", header=TRUE, row.names=1)
count_BoMac_samples <- read.table("bosRawCounts.txt", header=TRUE, row.names=1)
count_THP1_samples <- count_THP1[, dataTHP1$sample_name]
count_BoMac_samples <- count_BoMac_samples[, dataBo$sample_name]
rm(count_THP1)
combined_conds <- paste(dataTHP1$VIRULENCE, dataTHP1$infection_date, sep = "_")
f_thp1 <- count_THP1_samples
f_bomac <- count_BoMac_samples

filt_count_THP1_samples <- HTSFilter(count_THP1_samples, conds = combined_conds,plot = F)
combined_conds <- paste(dataBo$VIRULENCE, dataBo$infection_date, sep = "_")
filt_count_BoMac_samples <- HTSFilter(count_BoMac_samples, conds = combined_conds,plot = F)
f_thp1 <- filt_count_THP1_samples$filteredData
f_bomac <- filt_count_BoMac_samples$filteredData
#all(colnames(f_thp1) == rownames(dataTHP1))
#all(colnames(f_bomac) == rownames(dataBo))
rm(filt_count_THP1_samples)
rm(filt_count_BoMac_samples)
rm(count_THP1_samples)
rm(count_BoMac_samples)
dataTHP1$infection_date<-as.factor(dataTHP1$infection_date)
dataBo$infection_date<-as.factor(dataBo$infection_date)

dds_thp1 <- create_and_analyze_DESeqDataSet(f_thp1, dataTHP1, "~ VIRULENCE + infection_date")
dds_bomac <- create_and_analyze_DESeqDataSet(f_bomac, dataBo, "~ VIRULENCE + infection_date")
rm(f_thp1)
rm(f_bomac)
```

```{r functions2, echo = FALSE, message=FALSE, warning=FALSE}
analyze_and_annotate <- function(dds, contrast, mart_dataset = "hsapiens_gene_ensembl") {
  resultado <- DESeq2::results(dds, contrast = contrast)
  
  # Convertir a dataframe y ordenar por p-value
  resultadoOrdenado <- data.frame(resultado)
  resultadoOrdenado <- resultadoOrdenado[order(resultadoOrdenado$padj),]
  
  resultadoOrdenado$color <- ifelse(resultadoOrdenado$log2FoldChange > 1 & -log10(resultadoOrdenado$padj) > 1.3, "red",
                                    ifelse(resultadoOrdenado$log2FoldChange < -1 & -log10(resultadoOrdenado$padj) > 1.3, "blue", "grey"))
  
  ensembl <- useMart("ensembl", dataset = mart_dataset)
  
  genes_info <- getBM(attributes = c('ensembl_gene_id', 'external_gene_name', 'description'),
                      filters = 'ensembl_gene_id',
                      values = rownames(resultadoOrdenado),
                      mart = ensembl)
  
  genes_df <- merge(resultadoOrdenado, genes_info, by.x = 'row.names', by.y = 'ensembl_gene_id', all.x = TRUE)
  
  return(genes_df)
}

format_number <- function(x) {
  sapply(x, function(y) {
    if (is.na(y)) {
      return(NA)
    } else if (abs(y) < .Machine$double.eps ^ 0.5) {
      return(sprintf("%.5f", 0))  # Cero con 5 decimales
    } else if (abs(y) < 1e-3 || abs(y) > 1e3) {
      return(format(y, scientific = TRUE, digits = 5))  # NotaciÃ³n cientÃ­fica para valores muy pequeÃ±os o grandes
    } else {
      return(sprintf("%.5f", y))  # Formato normal con 5 decimales para otros valores
    }
  })
}

analyze_filter_export_and_save_table <- function(dds, groups, gene_id, file_name, csv_title, excel_title) {
    # AnÃ¡lisis y anotaciÃ³n
    bgenes_df <- analyze_and_annotate(dds, groups, gene_id)
    
    # Filtrado de datos
    bgenes_up <- filter(bgenes_df, color == 'red')
    bgenes_down <- filter(bgenes_df, color == 'blue')
    
    # Exportar a XLSX
    write_xlsx(bgenes_df, paste0(file_name, ".xlsx"))

    # Crear un nuevo dataframe
    bgenes_datatable <- data.frame(bgenes_df$Row.names, bgenes_df$external_gene_name, bgenes_df$padj, bgenes_df$log2FoldChange, bgenes_df$color)
    colnames(bgenes_datatable)<-c("ID","Gene","Padj","log2FoldChange","color")
    bgenes_datatable$Padj <- format_number(bgenes_datatable$Padj)
    bgenes_datatable$log2FoldChange <- format_number(bgenes_datatable$log2FoldChange)

    # Mostrar el datatable
    a<-datatable(bgenes_datatable, 
              extensions = 'Buttons', 
              options = list(
                  dom = 'Bfrtip',
                  buttons = list(
                      list(extend = 'csv', title = csv_title),
                      list(extend = 'excel', title = excel_title),
                      'pdf',
                      'print',
                      'copy'
                  ),
                  pageLength = 10,
                  autoWidth = TRUE,
                  ordering = TRUE,
                  lengthMenu = list(c(5, 10, 15, -1), c('5', '10', '15', 'All'))
              ),
              class = 'table table-striped table-bordered',
              filter = 'top',
          rownames = FALSE)

    # Devolver los objetos
    return(list(up = bgenes_up, down = bgenes_down, all_data = bgenes_df, dtable=a))
}
create_interactive_volcanoplot <- function(data) {
  g <- ggplot(data, aes(x = log2FoldChange, y = -log10(padj), color = color, text = external_gene_name)) +
    geom_point(alpha = 0.5) +
    theme_few() +
    labs(title = "", x = "Log2 Fold Change", y = "-Log10 p-value") +
    scale_color_identity() +
    geom_vline(xintercept = c(-1, 1), linetype = "dashed", color = "darkgrey") +
    geom_hline(yintercept = 1.3, linetype = "dashed", color = "darkgrey")

  ggplotly(g, tooltip = "text")
}
perform_analysis <- function(enrichment_results, orgdb, ont) {
  # Filtrar resultados por valor p ajustado
  filter_thp1bpdwn <- enrichment_results %>%
    filter(p.adjust < 0.05)
  
  # Calcular la matriz de similitud
  simMatrix <- calculateSimMatrix(filter_thp1bpdwn$ID,
                                  orgdb = orgdb,
                                  ont = ont,
                                  method = "Rel")
  
  # Convertir q-values a scores
  scores <- setNames(-log10(filter_thp1bpdwn$qvalue), filter_thp1bpdwn$ID)
  
  # Reducir la matriz de similitud
  reducedTerms <- reduceSimMatrix(simMatrix,
                                  scores,
                                  threshold = 0.7,
                                  orgdb = orgdb)
  
  # Devolver una lista con los resultados
  return(list(simMatrix = simMatrix, scores = scores, reducedTerms = reducedTerms))
}
```
### DEGs Table Results
:::: {.panel-tabset}
## THP-1
::: {.panel-tabset}
### High vs Low
```{r tableHvL, echo=FALSE, message=FALSE, warning=FALSE}
resultados_HvL <- analyze_filter_export_and_save_table(
    dds = dds_thp1,
    groups = c("VIRULENCE", "High", "Low"),
    gene_id = "hsapiens_gene_ensembl",
    file_name = "THP1_HvL",
    csv_title = "THP1_HvL_html",
    excel_title = "THP1_HvL_html")
resultados_HvL$dtable
```
### Medium vs High
```{r tableMvH, echo=FALSE, message=FALSE, warning=FALSE}
resultados_MvH <- analyze_filter_export_and_save_table(
    dds = dds_thp1,
    groups = c("VIRULENCE", "Medium", "High"),
    gene_id = "hsapiens_gene_ensembl",
    file_name = "THP1_MvH",
    csv_title = "THP1_MvH_html",
    excel_title = "THP1_MvH_html")
resultados_MvH$dtable
```
### Medium vs Low
```{r tableA1C, echo=FALSE, message=FALSE, warning=FALSE}
resultados_MvL <- analyze_filter_export_and_save_table(
    dds = dds_thp1,
    groups = c("VIRULENCE", "Medium", "Low"),
    gene_id = "hsapiens_gene_ensembl",
    file_name = "THP1_MvL",
    csv_title = "THP1_MvL_html",
    excel_title = "THP1_MvL_html")
resultados_MvL$dtable
```
:::
## BoMac
::: {.panel-tabset}
### High vs Low
```{r btableHvL, echo=FALSE, message=FALSE, warning=FALSE}
bresultados_HvL <- analyze_filter_export_and_save_table(
    dds = dds_bomac,
    groups = c("VIRULENCE", "High", "Low"),
    gene_id = "btaurus_gene_ensembl",
    file_name = "BoMac_HvL",
    csv_title = "BoMac_HvL_html",
    excel_title = "BoMac_HvL_html")
bresultados_HvL$dtable
```
### Medium vs High
```{r btableMvH, echo=FALSE, message=FALSE, warning=FALSE}
bresultados_MvH <- analyze_filter_export_and_save_table(
    dds = dds_bomac,
    groups = c("VIRULENCE", "Medium", "High"),
    gene_id = "btaurus_gene_ensembl",
    file_name = "BoMac_MvH",
    csv_title = "BoMac_MvH_html",
    excel_title = "BoMac_MvH_html")
bresultados_MvH$dtable
```
### Medium vs Low
```{r btableA1C, echo=FALSE, message=FALSE, warning=FALSE}
bresultados_MvL <- analyze_filter_export_and_save_table(
    dds = dds_bomac,
    groups = c("VIRULENCE", "Medium", "Low"),
    gene_id = "btaurus_gene_ensembl",
    file_name = "BoMac_MvL",
    csv_title = "BoMac_MvL_html",
    excel_title = "BoMac_MvL_html")
bresultados_MvL$dtable
```
:::
::::
### Volcano Plots
:::: {.panel-tabset}
## THP-1
::: {.panel-tabset}
### High vs Low
```{r vplotHvL, echo=FALSE, message=FALSE, warning=FALSE}
create_interactive_volcanoplot(resultados_HvL$all_data)
```
### Medium vs High
```{r vplotMvH, echo=FALSE, message=FALSE, warning=FALSE}
create_interactive_volcanoplot(resultados_MvH$all_data)
```
### Medium vs Low
```{r vplotMvL, echo=FALSE, message=FALSE, warning=FALSE}
create_interactive_volcanoplot(resultados_MvL$all_data)
```
:::
## BoMac
::: {.panel-tabset}
### High vs Low
```{r BvplotHvL, echo=FALSE, message=FALSE, warning=FALSE}
create_interactive_volcanoplot(bresultados_HvL$all_data)
```
### Medium vs High
```{r BvplotMvH, echo=FALSE, message=FALSE, warning=FALSE}
create_interactive_volcanoplot(bresultados_MvH$all_data)
```
### Medium vs Low
```{r BvplotMvL, echo=FALSE, message=FALSE, warning=FALSE}
create_interactive_volcanoplot(bresultados_MvL$all_data)
```
:::
::::
```{r enrich_kegg, echo = FALSE}
enrich_kegg <- function(all_up_thp1, org_db, organism_code) {
  # Convertir ENSEMBL IDs a Entrez IDs
  entrez_genes <- AnnotationDbi::mapIds(org_db, 
                                        keys = all_up_thp1, 
                                        column = "ENTREZID", 
                                        keytype = "ENSEMBL", 
                                        multiVals = "first")
  entrez_genes <- entrez_genes[!is.na(entrez_genes)]

  # Enriquecimiento KEGG
  kegg_result <- clusterProfiler::enrichKEGG(
    gene         = entrez_genes,
    organism     = organism_code,
    keyType      = "kegg",
    pvalueCutoff = 0.05,
    qvalueCutoff = 0.2
  )

  # Obtener todos los identificadores Entrez y convertirlos a nombres de genes
  all_entrez_ids <- unique(unlist(strsplit(as.character(kegg_result@result$geneID), "/")))
  entrez2name <- AnnotationDbi::select(org_db, 
                                       keys = all_entrez_ids, 
                                       columns = c("ENTREZID", "SYMBOL"), 
                                       keytype = "ENTREZID")
  entrez2name <- entrez2name[!duplicated(entrez2name$ENTREZID), ]

  # FunciÃ³n para convertir Entrez IDs a nombres de genes
  convert_entrez_to_name <- function(entrez_ids_str) {
    entrez_ids <- strsplit(entrez_ids_str, "/")[[1]]
    gene_names <- entrez2name$SYMBOL[match(entrez_ids, entrez2name$ENTREZID)]
    return(paste(gene_names, collapse = "/"))
  }

  # Aplicar la conversiÃ³n a los resultados de KEGG
  kegg_result@result$geneID <- sapply(as.character(kegg_result@result$geneID), convert_entrez_to_name)

  # Devolver el resultado
  return(kegg_result)
}
```

```{r enrich_GO, echo = FALSE}
enrich_GO_and_convert <- function(id_genes, org_db, ontology) {
    # Convertir ENSEMBL IDs a Entrez IDs
  entrez_genes <- AnnotationDbi::mapIds(org_db, 
                                        keys = id_genes, 
                                        column = "ENTREZID", 
                                        keytype = "ENSEMBL", 
                                        multiVals = "first")
  entrez_genes <- entrez_genes[!is.na(entrez_genes)]
  
  # Enriquecimiento GO
  ego_GO <- enrichGO(
    gene          = entrez_genes,
    OrgDb         = org_db,
    keyType       = "ENTREZID",
    ont           = ontology,  # BP, CC o MF
    pAdjustMethod = "BH",
    pvalueCutoff  = 0.05,
    qvalueCutoff  = 0.2    
  )

  # Obtiene los identificadores Entrez Ãºnicos de los resultados
  all_entrez_ids <- unique(unlist(strsplit(as.character(ego_GO@result$geneID), "/")))

  # Mapea los identificadores Entrez a los nombres de los genes
  entrez2name <- AnnotationDbi::select(org_db, 
                                       keys = all_entrez_ids, 
                                       columns = c("ENTREZID", "SYMBOL"), 
                                       keytype = "ENTREZID")

  # Elimina duplicados
  entrez2name <- entrez2name[!duplicated(entrez2name$ENTREZID), ]

  # FunciÃ³n para convertir identificadores Entrez en nombres de genes
  convert_entrez_to_name <- function(entrez_ids_str) {
    entrez_ids <- strsplit(entrez_ids_str, "/")[[1]]
    gene_names <- entrez2name$SYMBOL[match(entrez_ids, entrez2name$ENTREZID)]
    return(paste(gene_names, collapse = "/"))
  }

  # Reemplaza los valores en la columna geneID con los nombres de los genes
  ego_GO@result$geneID <- sapply(as.character(ego_GO@result$geneID), convert_entrez_to_name)

  return(ego_GO)
}

create_datatable <- function(data, title) {
  # Check if the columns exist and apply format_number if they do
  if("pvalue" %in% names(data)) {
    data$pvalue <- format_number(data$pvalue)
  }
  if("p.adjust" %in% names(data)) {
    data$p.adjust <- format_number(data$p.adjust)
  }
  if("qvalue" %in% names(data)) {
    data$qvalue <- format_number(data$qvalue)
  }

  a<-datatable(data, 
            extensions = 'Buttons', 
            options = list(
                dom = 'Bfrtip',
                buttons = list(
                    list(extend = 'csv', title = paste0(title, "_common")),
                    list(extend = 'excel', title = paste0(title, "_common")),
                    'pdf',
                    'print',
                    'copy'
                ),
                pageLength = 3,
                autoWidth = TRUE,
                ordering = TRUE,
                lengthMenu = list(c(5, 10, 15, -1), c('5', '10', '15', 'All'))
            ),
            class = 'table table-striped table-bordered',
            filter = 'top',
          rownames = FALSE)
  return(a)
}
```

### Gene Enrichment Results
::::::: {.panel-tabset}
# THP-1
:::::: {.panel-tabset}
## High vs Low
::::: {.panel-tabset}
### Up-regulated
:::: {.panel-tabset}
#### KEGG Pathways
::: {.panel-tabset}
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
result_upthp1HvL<-enrich_kegg(resultados_HvL$up$Row.names, org.Hs.eg.db, 'hsa')
```
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
dotplot(result_upthp1HvL)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(result_upthp1HvL@result, "kegg_upthp1L6vC")
```
:::
#### Biological Process
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
bp_upthp1HvL<-enrich_GO_and_convert(resultados_HvL$up$Row.names, org.Hs.eg.db, "BP")
dotplot(bp_upthp1HvL)
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
rrBo_up_BP_thp1_HvL<-perform_analysis(bp_upthp1HvL@result, "org.Hs.eg.db", "BP")
wordcloudPlot(rrBo_up_BP_thp1_HvL$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
scatterPlot(rrBo_up_BP_thp1_HvL$simMatrix, rrBo_up_BP_thp1_HvL$reducedTerms)
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
treemapPlot(rrBo_up_BP_thp1_HvL$reducedTerms)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(bp_upthp1HvL@result, "BP_upthp1HvL")
```
:::
#### Molecular Function
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
mf_upthp1HvL<-enrich_GO_and_convert(resultados_HvL$up$Row.names, org.Hs.eg.db, "MF")
dotplot(mf_upthp1HvL)
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
rrBo_up_MF_thp1_HvL<-perform_analysis(mf_upthp1HvL@result, "org.Hs.eg.db", "MF")
wordcloudPlot(rrBo_up_MF_thp1_HvL$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
scatterPlot(rrBo_up_MF_thp1_HvL$simMatrix, rrBo_up_MF_thp1_HvL$reducedTerms)
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
treemapPlot(rrBo_up_MF_thp1_HvL$reducedTerms)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(mf_upthp1HvL@result, "MF_upthp1HvL")
```
:::
#### Cellular Component
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
cc_upthp1HvL<-enrich_GO_and_convert(resultados_HvL$up$Row.names, org.Hs.eg.db, "CC")
dotplot(cc_upthp1HvL)
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
rrBo_up_CC_thp1_HvL<-perform_analysis(cc_upthp1HvL@result, "org.Hs.eg.db", "CC")
wordcloudPlot(rrBo_up_CC_thp1_HvL$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
scatterPlot(rrBo_up_CC_thp1_HvL$simMatrix, rrBo_up_CC_thp1_HvL$reducedTerms)
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
treemapPlot(rrBo_up_CC_thp1_HvL$reducedTerms)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(cc_upthp1HvL@result, "CC_upthp1HvL")
```
:::
::::
### Down-regulated
:::: {.panel-tabset}
#### KEGG Pathways
::: {.panel-tabset}
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
result_downthp1HvL<-enrich_kegg(resultados_HvL$down$Row.names, org.Hs.eg.db, 'hsa')
```
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
dotplot(result_downthp1HvL)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(result_downthp1HvL@result, "kegg_downthp1HvL")
```
:::
#### Biological Process
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
bp_downthp1HvL<-enrich_GO_and_convert(resultados_HvL$down$Row.names, org.Hs.eg.db, "BP")
dotplot(bp_downthp1HvL)
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
rrBo_down_BP_thp1_HvL<-perform_analysis(bp_downthp1HvL@result, "org.Hs.eg.db", "BP")
wordcloudPlot(rrBo_down_BP_thp1_HvL$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
scatterPlot(rrBo_down_BP_thp1_HvL$simMatrix, rrBo_down_BP_thp1_HvL$reducedTerms)
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
treemapPlot(rrBo_down_BP_thp1_HvL$reducedTerms)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(bp_downthp1HvL@result, "BP_downthp1HvL")
```
:::
#### Molecular Function
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
mf_downthp1HvL<-enrich_GO_and_convert(resultados_HvL$down$Row.names, org.Hs.eg.db, "MF")
dotplot(mf_downthp1HvL)
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
rrBo_down_MF_thp1_HvL<-perform_analysis(mf_downthp1HvL@result, "org.Hs.eg.db", "MF")
wordcloudPlot(rrBo_down_MF_thp1_HvL$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
scatterPlot(rrBo_down_MF_thp1_HvL$simMatrix, rrBo_down_MF_thp1_HvL$reducedTerms)
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
treemapPlot(rrBo_down_MF_thp1_HvL$reducedTerms)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(mf_downthp1HvL@result, "MF_downthp1HvL")
```
:::
#### Cellular Component
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
cc_downthp1HvL<-enrich_GO_and_convert(resultados_HvL$down$Row.names, org.Hs.eg.db, "CC")
dotplot(cc_downthp1HvL)
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
rrBo_down_CC_thp1_HvL<-perform_analysis(cc_downthp1HvL@result, "org.Hs.eg.db", "CC")
wordcloudPlot(rrBo_down_CC_thp1_HvL$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
scatterPlot(rrBo_down_CC_thp1_HvL$simMatrix, rrBo_down_CC_thp1_HvL$reducedTerms)
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
treemapPlot(rrBo_down_CC_thp1_HvL$reducedTerms)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(cc_downthp1HvL@result, "CC_downthp1L6vC")
```
:::
::::
:::::
## Medium vs High
::::: {.panel-tabset}
### Up-regulated
:::: {.panel-tabset}
#### KEGG Pathways
::: {.panel-tabset}
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
result_upthp1MvH<-enrich_kegg(resultados_MvH$up$Row.names, org.Hs.eg.db, 'hsa')
```
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
dotplot(result_upthp1MvH)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(result_upthp1MvH@result, "kegg_upthp1MvH")
```
:::
#### Biological Process
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
bp_upthp1MvH<-enrich_GO_and_convert(resultados_MvH$up$Row.names, org.Hs.eg.db, "BP")
dotplot(bp_upthp1MvH)
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
rrBo_up_BP_thp1_MvH<-perform_analysis(bp_upthp1MvH@result, "org.Hs.eg.db", "BP")
wordcloudPlot(rrBo_up_BP_thp1_MvH$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
scatterPlot(rrBo_up_BP_thp1_MvH$simMatrix, rrBo_up_BP_thp1_MvH$reducedTerms)
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
treemapPlot(rrBo_up_BP_thp1_MvH$reducedTerms)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(bp_upthp1MvH@result, "BP_upthp1MvH")
```
:::
#### Molecular Function
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
mf_upthp1MvH<-enrich_GO_and_convert(resultados_MvH$up$Row.names, org.Hs.eg.db, "MF")
dotplot(mf_upthp1MvH)
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
rrBo_up_MF_thp1_MvH<-perform_analysis(mf_upthp1MvH@result, "org.Hs.eg.db", "MF")
wordcloudPlot(rrBo_up_MF_thp1_MvH$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
scatterPlot(rrBo_up_MF_thp1_MvH$simMatrix, rrBo_up_MF_thp1_MvH$reducedTerms)
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
treemapPlot(rrBo_up_MF_thp1_MvH$reducedTerms)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(mf_upthp1MvH@result, "MF_upthp1MvH")
```
:::
#### Cellular Component
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
cc_upthp1MvH<-enrich_GO_and_convert(resultados_MvH$up$Row.names, org.Hs.eg.db, "CC")
dotplot(cc_upthp1MvH)
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
rrBo_up_CC_thp1_MvH<-perform_analysis(cc_upthp1MvH@result, "org.Hs.eg.db", "CC")
wordcloudPlot(rrBo_up_CC_thp1_MvH$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
scatterPlot(rrBo_up_CC_thp1_MvH$simMatrix, rrBo_up_CC_thp1_MvH$reducedTerms)
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
treemapPlot(rrBo_up_CC_thp1_MvH$reducedTerms)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(cc_upthp1MvH@result, "CC_upthp1MvH")
```
:::
::::
### Down-regulated
:::: {.panel-tabset}
#### KEGG Pathways
::: {.panel-tabset}
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#result_downthp1MvH<-enrich_kegg(resultados_MvH$down$Row.names, org.Hs.eg.db, 'hsa')
```
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#dotplot(result_downthp1MvH)
print("No enrichment results")
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#create_datatable(result_downthp1L5vC@result, "kegg_downthp1L5vC")
print("No enrichment results")
```
:::
#### Biological Process
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#bp_downthp1MvH<-enrich_GO_and_convert(resultados_MvH$down$Row.names, org.Hs.eg.db, "BP")
#dotplot(bp_downthp1MvH)
print("No enrichment results")
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#rrBo_down_BP_thp1_MvH<-perform_analysis(bp_downthp1MvH@result, "org.Hs.eg.db", "BP")
#wordcloudPlot(rrBo_down_BP_thp1_MvH$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
print("No enrichment results")
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#scatterPlot(rrBo_down_BP_thp1_L5$simMatrix, rrBo_down_BP_thp1_L5$reducedTerms)
print("No enrichment results")
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#treemapPlot(rrBo_down_BP_thp1_L5$reducedTerms)
print("No enrichment results")
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#create_datatable(bp_downthp1L5vC@result, "BP_downthp1L5vC")
print("No enrichment results")
```
:::
#### Molecular Function
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#mf_downthp1MvH<-enrich_GO_and_convert(resultados_MvH$down$Row.names, org.Hs.eg.db, "MF")
#dotplot(mf_downthp1MvH)
print("No enrichment results")
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#rrBo_down_MF_thp1_MvH<-perform_analysis(mf_downthp1MvH@result, "org.Hs.eg.db", "MF")
#wordcloudPlot(rrBo_down_MF_thp1_MvH$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
print("No enrichment results")
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#scatterPlot(rrBo_down_MF_thp1_L5$simMatrix, rrBo_down_MF_thp1_L5$reducedTerms)
print("No enrichment results")
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#treemapPlot(rrBo_down_MF_thp1_L5$reducedTerms)
print("No enrichment results")
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#create_datatable(mf_downthp1L5vC@result, "MF_downthp1L5vC")
print("No enrichment results")
```
:::
#### Cellular Component
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#cc_downthp1MvH<-enrich_GO_and_convert(resultados_MvH$down$Row.names, org.Hs.eg.db, "CC")
#dotplot(cc_downthp1MvH)
print("No enrichment results")
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#rrBo_down_CC_thp1_L5<-perform_analysis(cc_downthp1L5vC@result, "org.Hs.eg.db", "CC")
#wordcloudPlot(rrBo_down_CC_thp1_L5$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
print("No enrichment results")
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#scatterPlot(rrBo_down_CC_thp1_L5$simMatrix, rrBo_down_CC_thp1_L5$reducedTerms)
print("No enrichment results")
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#treemapPlot(rrBo_down_CC_thp1_L5$reducedTerms)
print("No enrichment results")
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#create_datatable(cc_downthp1L5vC@result, "CC_downthp1L5vC")
print("No enrichment results")
```
:::
::::
:::::
## Medium vs Low
::::: {.panel-tabset}
### Up-regulated
:::: {.panel-tabset}
#### KEGG Pathways
::: {.panel-tabset}
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
result_upthp1MvL<-enrich_kegg(resultados_MvL$up$Row.names, org.Hs.eg.db, 'hsa')
```
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
dotplot(result_upthp1MvL)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(result_upthp1MvL@result, "kegg_upthp1MvL")
```
:::
#### Biological Process
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
bp_upthp1MvL<-enrich_GO_and_convert(resultados_MvL$up$Row.names, org.Hs.eg.db, "BP")
dotplot(bp_upthp1MvL)
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
rrBo_up_BP_thp1_MvL<-perform_analysis(bp_upthp1MvL@result, "org.Hs.eg.db", "BP")
wordcloudPlot(rrBo_up_BP_thp1_MvL$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
scatterPlot(rrBo_up_BP_thp1_MvL$simMatrix, rrBo_up_BP_thp1_MvL$reducedTerms)
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
treemapPlot(rrBo_up_BP_thp1_MvL$reducedTerms)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(bp_upthp1MvL@result, "BP_upthp1MvL")
```
:::
#### Molecular Function
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
mf_upthp1MvL<-enrich_GO_and_convert(resultados_MvL$up$Row.names, org.Hs.eg.db, "MF")
dotplot(mf_upthp1MvL)
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
rrBo_up_MF_thp1_MvL<-perform_analysis(mf_upthp1MvL@result, "org.Hs.eg.db", "MF")
wordcloudPlot(rrBo_up_MF_thp1_MvL$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
scatterPlot(rrBo_up_MF_thp1_MvL$simMatrix, rrBo_up_MF_thp1_MvL$reducedTerms)
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
treemapPlot(rrBo_up_MF_thp1_MvL$reducedTerms)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(mf_upthp1MvL@result, "MF_upthp1MvL")
```
:::
#### Cellular Component
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
cc_upthp1MvL<-enrich_GO_and_convert(resultados_MvL$up$Row.names, org.Hs.eg.db, "CC")
dotplot(cc_upthp1MvL)
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
rrBo_up_CC_thp1_MvL<-perform_analysis(cc_upthp1MvL@result, "org.Hs.eg.db", "CC")
wordcloudPlot(rrBo_up_CC_thp1_MvL$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
scatterPlot(rrBo_up_CC_thp1_MvL$simMatrix, rrBo_up_CC_thp1_MvL$reducedTerms)
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
treemapPlot(rrBo_up_CC_thp1_MvL$reducedTerms)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(cc_upthp1MvL@result, "CC_upthp1MvL")
```
:::
::::
### Down-regulated
:::: {.panel-tabset}
#### KEGG Pathways
::: {.panel-tabset}
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
result_downthp1MvL<-enrich_kegg(resultados_MvL$down$Row.names, org.Hs.eg.db, 'hsa')
```
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
dotplot(result_downthp1MvL)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(result_downthp1MvL@result, "kegg_downthp1MvL")
```
:::
#### Biological Process
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
bp_downthp1MvL<-enrich_GO_and_convert(resultados_MvL$down$Row.names, org.Hs.eg.db, "BP")
dotplot(bp_downthp1MvL)
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
rrBo_down_BP_thp1_MvL<-perform_analysis(bp_downthp1MvL@result, "org.Hs.eg.db", "BP")
wordcloudPlot(rrBo_down_BP_thp1_MvL$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
scatterPlot(rrBo_down_BP_thp1_MvL$simMatrix, rrBo_down_BP_thp1_MvL$reducedTerms)
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
treemapPlot(rrBo_down_BP_thp1_MvL$reducedTerms)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(bp_downthp1MvL@result, "BP_downthp1MvL")
```
:::
#### Molecular Function
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
mf_downthp1MvL<-enrich_GO_and_convert(resultados_MvL$down$Row.names, org.Hs.eg.db, "MF")
dotplot(mf_downthp1MvL)
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
rrBo_down_MF_thp1_MvL<-perform_analysis(mf_downthp1MvL@result, "org.Hs.eg.db", "MF")
wordcloudPlot(rrBo_down_MF_thp1_MvL$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
scatterPlot(rrBo_down_MF_thp1_MvL$simMatrix, rrBo_down_MF_thp1_MvL$reducedTerms)
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
treemapPlot(rrBo_down_MF_thp1_MvL$reducedTerms)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(mf_downthp1MvL@result, "MF_downthp1MvL")
```
:::
#### Cellular Component
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
cc_downthp1MvL<-enrich_GO_and_convert(resultados_MvL$down$Row.names, org.Hs.eg.db, "CC")
dotplot(cc_downthp1MvL)
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
rrBo_down_CC_thp1_MvL<-perform_analysis(cc_downthp1MvL@result, "org.Hs.eg.db", "CC")
wordcloudPlot(rrBo_down_CC_thp1_MvL$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
scatterPlot(rrBo_down_CC_thp1_MvL$simMatrix, rrBo_down_CC_thp1_MvL$reducedTerms)
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
treemapPlot(rrBo_down_CC_thp1_MvL$reducedTerms)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(cc_downthp1MvL@result, "CC_downthp1MvL")
```
:::
::::
:::::
::::::
# BoMac
:::::: {.panel-tabset}
## High vs Low
::::: {.panel-tabset}
### Up-regulated
:::: {.panel-tabset}
#### KEGG Pathways
::: {.panel-tabset}
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
result_upbomacHvL<-enrich_kegg(bresultados_HvL$up$Row.names, org.Bt.eg.db, 'bta')
```
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
dotplot(result_upbomacHvL)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(result_upbomacHvL@result, "kegg_upbomacHvL")
```
:::
#### Biological Process
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#bp_upbomacHvL<-enrich_GO_and_convert(bresultados_HvL$up$Row.names, org.Bt.eg.db, "BP")
#dotplot(bp_upbomacHvL)
print("No enrichment result")
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#rrBo_up_BP_bomac_HvL<-perform_analysis(bp_upbomacHvL@result, "org.Bt.eg.db", "BP")
#wordcloudPlot(rrBo_up_BP_bomac_HvL$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
print("No enrichment result")
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#scatterPlot(rrBo_up_BP_bomac_HvL$simMatrix, rrBo_up_BP_bomac_HvL$reducedTerms)
print("No enrichment result")
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#treemapPlot(rrBo_up_BP_bomac_HvL$reducedTerms)
print("No enrichment result")
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#create_datatable(bp_upbomacHvL@result, "BP_upbomacHvL")
print("No enrichment result")
```
:::
#### Molecular Function
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#mf_upbomacHvL<-enrich_GO_and_convert(bresultados_HvL$up$Row.names, org.Bt.eg.db, "MF")
#dotplot(mf_upbomacHvL)
print("No enrichment result")
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#rrBo_up_MF_bomac_HvL<-perform_analysis(mf_upbomacHvL@result, "org.Bt.eg.db", "MF")
#wordcloudPlot(rrBo_up_MF_bomac_HvL$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
print("No enrichment result")
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#scatterPlot(rrBo_up_MF_bomac_HvL$simMatrix, rrBo_up_MF_bomac_HvL$reducedTerms)
print("No enrichment result")
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#treemapPlot(rrBo_up_MF_bomac_HvL$reducedTerms)
print("No enrichment result")
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#create_datatable(mf_upbomacHvL@result, "MF_upbomacHvL")
print("No enrichment result")
```
:::
#### Cellular Component
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
cc_upbomacHvL<-enrich_GO_and_convert(bresultados_HvL$up$Row.names, org.Bt.eg.db, "CC")
dotplot(cc_upbomacHvL)
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
rrBo_up_CC_bomac_HvL<-perform_analysis(cc_upbomacHvL@result, "org.Bt.eg.db", "CC")
wordcloudPlot(rrBo_up_CC_bomac_HvL$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
scatterPlot(rrBo_up_CC_bomac_HvL$simMatrix, rrBo_up_CC_bomac_HvL$reducedTerms)
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
treemapPlot(rrBo_up_CC_bomac_HvL$reducedTerms)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(cc_upbomacHvL@result, "CC_upbomacHvL")
```
:::
::::
### Down-regulated
:::: {.panel-tabset}
#### KEGG Pathways
::: {.panel-tabset}
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
result_downbomacHvL<-enrich_kegg(bresultados_HvL$down$Row.names, org.Bt.eg.db, 'bta')
```
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
dotplot(result_downbomacHvL)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(result_downbomacHvL@result, "kegg_downbomacHvL")
```
:::
#### Biological Process
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#bp_downbomacHvL<-enrich_GO_and_convert(bresultados_HvL$down$Row.names, org.Bt.eg.db, "BP")
#dotplot(bp_downbomacHvL)
print("No enrichment results")
```
:::
#### Molecular Function
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#mf_downbomacHvL<-enrich_GO_and_convert(bresultados_HvL$down$Row.names, org.Bt.eg.db, "MF")
#dotplot(mf_downbomacHvL)
print("No enrichment results")
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#rrBo_down_MF_bomac_L6<-perform_analysis(mf_downbomacL6vC@result, "org.Bt.eg.db", "MF")
#wordcloudPlot(rrBo_down_MF_bomac_L6$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
print("No enrichment results")
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#scatterPlot(rrBo_down_MF_bomac_L6$simMatrix, rrBo_down_MF_bomac_L6$reducedTerms)
print("No enrichment results")
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#treemapPlot(rrBo_down_MF_bomac_L6$reducedTerms)
print("No enrichment results")
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#create_datatable(mf_downbomacL6vC@result, "MF_downbomacL6vC")
print("No enrichment results")
```
:::
#### Cellular Component
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#cc_downbomacHvL<-enrich_GO_and_convert(bresultados_HvL$down$Row.names, org.Bt.eg.db, "CC")
#dotplot(cc_downbomacHvL)
print("No enrichment result")
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#rrBo_down_CC_bomac_L6<-perform_analysis(cc_downbomacL6vC@result, "org.Bt.eg.db", "CC")
#wordcloudPlot(rrBo_down_CC_bomac_L6$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
print("No enrichment result")
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#scatterPlot(rrBo_down_CC_bomac_L6$simMatrix, rrBo_down_CC_bomac_L6$reducedTerms)
print("No enrichment result")
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#treemapPlot(rrBo_down_CC_bomac_L6$reducedTerms)
print("No enrichment result")
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#create_datatable(cc_downbomacL6vC@result, "CC_downbomacL6vC")
print("No enrichment result")
```
:::
::::
:::::
## Medium vs High
::::: {.panel-tabset}
### Up-regulated
:::: {.panel-tabset}
#### KEGG Pathways
::: {.panel-tabset}
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
result_upbomacMvH<-enrich_kegg(bresultados_MvH$up$Row.names, org.Bt.eg.db, 'bta')
```
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
dotplot(result_upbomacMvH)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(result_upbomacMvH@result, "kegg_upbomacMvH")
```
:::
#### Biological Process
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
bp_upbomacMvH<-enrich_GO_and_convert(bresultados_MvH$up$Row.names, org.Bt.eg.db, "BP")
dotplot(bp_upbomacMvH)
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
rrBo_up_BP_bomac_MvH<-perform_analysis(bp_upbomacMvH@result, "org.Bt.eg.db", "BP")
wordcloudPlot(rrBo_up_BP_bomac_MvH$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
scatterPlot(rrBo_up_BP_bomac_MvH$simMatrix, rrBo_up_BP_bomac_MvH$reducedTerms)
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
treemapPlot(rrBo_up_BP_bomac_MvH$reducedTerms)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(bp_upbomacMvH@result, "BP_upbomacL5vC")
```
:::
#### Molecular Function
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#mf_upbomacMvH<-enrich_GO_and_convert(bresultados_MvH$up$Row.names, org.Bt.eg.db, "MF")
#dotplot(mf_upbomacMvH)
print("No enrichment results")
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#rrBo_up_MF_bomac_MvH<-perform_analysis(mf_upbomacMvH@result, "org.Bt.eg.db", "MF")
#wordcloudPlot(rrBo_up_MF_bomac_MvH$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
print("No enrichment results")
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#scatterPlot(rrBo_up_MF_bomac_MvH$simMatrix, rrBo_up_MF_bomac_MvH$reducedTerms)
print("No enrichment results")
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#treemapPlot(rrBo_up_MF_bomac_MvH$reducedTerms)
print("No enrichment results")
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#create_datatable(mf_upbomacMvH@result, "MF_upbomacL5vC")
print("No enrichment results")
```
:::
#### Cellular Component
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
cc_upbomacMvH<-enrich_GO_and_convert(bresultados_MvH$up$Row.names, org.Bt.eg.db, "CC")
dotplot(cc_upbomacMvH)
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
rrBo_up_CC_bomac_MvH<-perform_analysis(cc_upbomacMvH@result, "org.Bt.eg.db", "CC")
wordcloudPlot(rrBo_up_CC_bomac_MvH$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
scatterPlot(rrBo_up_CC_bomac_MvH$simMatrix, rrBo_up_CC_bomac_MvH$reducedTerms)
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
treemapPlot(rrBo_up_CC_bomac_MvH$reducedTerms)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(cc_upbomacMvH@result, "CC_upbomacMvH")
```
:::
::::
### Down-regulated
:::: {.panel-tabset}
#### KEGG Pathways
::: {.panel-tabset}
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
result_downbomacMvH<-enrich_kegg(bresultados_MvH$down$Row.names, org.Bt.eg.db, 'bta')
```
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
dotplot(result_downbomacMvH)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(result_downbomacMvH@result, "kegg_downbomacMvH")
```
:::
#### Biological Process
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#bp_downbomacMvH<-enrich_GO_and_convert(bresultados_MvH$down$Row.names, org.Bt.eg.db, "BP")
#dotplot(bp_downbomacMvH)
print("No enrichment results")
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#rrBo_down_BP_bomac_L5<-perform_analysis(bp_downbomacL5vC@result, "org.Bt.eg.db", "BP")
#wordcloudPlot(rrBo_down_BP_bomac_L5$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
print("No enrichment results")
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#scatterPlot(rrBo_down_BP_bomac_L5$simMatrix, rrBo_down_BP_bomac_L5$reducedTerms)
print("No enrichment results")
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#treemapPlot(rrBo_down_BP_bomac_L5$reducedTerms)
print("No enrichment results")
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#create_datatable(bp_downbomacL5vC@result, "BP_downbomacL5vC")
print("No enrichment results")
```
:::
#### Molecular Function
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
mf_downbomacMvH<-enrich_GO_and_convert(bresultados_MvH$down$Row.names, org.Bt.eg.db, "MF")
dotplot(mf_downbomacMvH)
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
rrBo_down_MF_bomac_MvH<-perform_analysis(mf_downbomacMvH@result, "org.Bt.eg.db", "MF")
wordcloudPlot(rrBo_down_MF_bomac_MvH$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
scatterPlot(rrBo_down_MF_bomac_MvH$simMatrix, rrBo_down_MF_bomac_MvH$reducedTerms)
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
treemapPlot(rrBo_down_MF_bomac_MvH$reducedTerms)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(mf_downbomacMvH@result, "MF_downbomacMvH")
```
:::
#### Cellular Component
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#cc_downbomacL5vC<-enrich_GO_and_convert(bresultados_MvH$down$Row.names, org.Bt.eg.db, "CC")
#dotplot(cc_downbomacMvH)
print("No enrichment results")
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#rrBo_down_CC_bomac_L5<-perform_analysis(cc_downbomacL5vC@result, "org.Bt.eg.db", "CC")
#wordcloudPlot(rrBo_down_CC_bomac_L5$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
print("No enrichment results")
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#scatterPlot(rrBo_down_CC_bomac_L5$simMatrix, rrBo_down_CC_bomac_L5$reducedTerms)
print("No enrichment results")
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#treemapPlot(rrBo_down_CC_bomac_L5$reducedTerms)
print("No enrichment results")
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#create_datatable(cc_downbomacL5vC@result, "CC_downbomacL5vC")
print("No enrichment results")
```
:::
::::
:::::
## Medium vs Low
::::: {.panel-tabset}
### Up-regulated
:::: {.panel-tabset}
#### KEGG Pathways
::: {.panel-tabset}
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
result_upbomacMvL<-enrich_kegg(bresultados_MvL$up$Row.names, org.Bt.eg.db, 'bta')
```
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
dotplot(result_upbomacMvL)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(result_upbomacMvL@result, "kegg_upbomacMvL")
```
:::
#### Biological Process
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
bp_upbomacMvL<-enrich_GO_and_convert(bresultados_MvL$up$Row.names, org.Bt.eg.db, "BP")
dotplot(bp_upbomacMvL)
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
rrBo_up_BP_bomac_MvL<-perform_analysis(bp_upbomacMvL@result, "org.Bt.eg.db", "BP")
wordcloudPlot(rrBo_up_BP_bomac_MvL$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
scatterPlot(rrBo_up_BP_bomac_MvL$simMatrix, rrBo_up_BP_bomac_MvL$reducedTerms)
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
treemapPlot(rrBo_up_BP_bomac_MvL$reducedTerms)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(bp_upbomacMvL@result, "BP_upbomacMvL")
```
:::
#### Molecular Function
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
mf_upbomacMvL<-enrich_GO_and_convert(bresultados_MvL$up$Row.names, org.Bt.eg.db, "MF")
dotplot(mf_upbomacMvL)
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
rrBo_up_MF_bomac_MvL<-perform_analysis(mf_upbomacMvL@result, "org.Bt.eg.db", "MF")
wordcloudPlot(rrBo_up_MF_bomac_MvL$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
scatterPlot(rrBo_up_MF_bomac_MvL$simMatrix, rrBo_up_MF_bomac_MvL$reducedTerms)
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
treemapPlot(rrBo_up_MF_bomac_MvL$reducedTerms)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(mf_upbomacMvL@result, "MF_upbomacMvL")
```
:::
#### Cellular Component
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
cc_upbomacMvL<-enrich_GO_and_convert(bresultados_MvL$up$Row.names, org.Bt.eg.db, "CC")
dotplot(cc_upbomacMvL)
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
rrBo_up_CC_bomac_MvL<-perform_analysis(cc_upbomacMvL@result, "org.Bt.eg.db", "CC")
wordcloudPlot(rrBo_up_CC_bomac_MvL$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
scatterPlot(rrBo_up_CC_bomac_MvL$simMatrix, rrBo_up_CC_bomac_MvL$reducedTerms)
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
treemapPlot(rrBo_up_CC_bomac_MvL$reducedTerms)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(cc_upbomacMvL@result, "CC_upbomacMvL")
```
:::
::::
### Down-regulated
:::: {.panel-tabset}
#### KEGG Pathways
::: {.panel-tabset}
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#result_downbomacMvL<-enrich_kegg(bresultados_MvL$down$Row.names, org.Bt.eg.db, 'bta')
```
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#dotplot(result_downbomacMvL)
print("No enrichment result")
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#create_datatable(result_downbomacMvL@result, "kegg_downbomacMvL")
print("No enrichment result")
```
:::
#### Biological Process
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#bp_downbomacMvL<-enrich_GO_and_convert(bresultados_MvL$down$Row.names, org.Bt.eg.db, "BP")
#dotplot(bp_downbomacMvL)
print("No enrichment result")
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#rrBo_down_BP_bomac_A1<-perform_analysis(bp_downbomacA1vC@result, "org.Bt.eg.db", "BP")
#wordcloudPlot(rrBo_down_BP_bomac_A1$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
print("No enrichment result")
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#scatterPlot(rrBo_down_BP_bomac_A1$simMatrix, rrBo_down_BP_bomac_A1$reducedTerms)
print("No enrichment result")
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#treemapPlot(rrBo_down_BP_bomac_A1$reducedTerms)
print("No enrichment result")
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#create_datatable(bp_downbomacA1vC@result, "BP_downbomacA1vC")
print("No enrichment result")
```
:::
#### Molecular Function
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
mf_downbomacMvL<-enrich_GO_and_convert(bresultados_MvL$down$Row.names, org.Bt.eg.db, "MF")
dotplot(mf_downbomacMvL)
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
rrBo_down_MF_bomac_MvL<-perform_analysis(mf_downbomacMvL@result, "org.Bt.eg.db", "MF")
wordcloudPlot(rrBo_down_MF_bomac_MvL$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#scatterPlot(rrBo_down_MF_bomac_MvL$simMatrix, rrBo_down_MF_bomac_MvL$reducedTerms)
print("Cannot be plotted")
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
treemapPlot(rrBo_down_MF_bomac_MvL$reducedTerms)
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(mf_downbomacMvL@result, "MF_downbomacMvL")
```
:::
#### Cellular Component
::: {.panel-tabset}
##### TOP Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#cc_downbomacMvL<-enrich_GO_and_convert(bresultados_MvL$down$Row.names, org.Bt.eg.db, "CC")
#dotplot(cc_downbomacMvL)
print("No enrichment results")
```
##### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#rrBo_down_CC_bomac_MvL<-perform_analysis(cc_downbomacMvL@result, "org.Bt.eg.db", "CC")
#wordcloudPlot(rrBo_down_CC_bomac_MvL$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
print("No enrichment results")
```
##### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#scatterPlot(rrBo_down_CC_bomac_MvL$simMatrix, rrBo_down_CC_bomac_MvL$reducedTerms)
print("No enrichment results")
```
##### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#treemapPlot(rrBo_down_CC_bomac_MvL$reducedTerms)
print("No enrichment results")
```
##### GE table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#create_datatable(cc_downbomacMvL@result, "CC_downbomacMvL")
print("No enrichment results")
```
:::
::::
:::::
::::::
:::::::
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
procesar_y_generar_venn <- function(result_upthp1HvL, result_upbomacHvL) {
    # Filtrar los resultados
    res_result_upthp1HvL <- result_upthp1HvL@result %>%
        filter(p.adjust < 0.05)
    res_result_upbomacHvL <- result_upbomacHvL@result %>%
        filter(p.adjust < 0.05)

    # Crear listas para el diagrama de Venn
    up_kegg_HvL <- list(
        `THP-1` = gsub("hsa", "", res_result_upthp1HvL$ID),
        `BoMac` = gsub("bta", "", res_result_upbomacHvL$ID)
    )

    # Comprobaciones bÃ¡sicas
    if(any(sapply(up_kegg_HvL, is.null) | sapply(up_kegg_HvL, length) == 0)) {
        stop("Uno o mÃ¡s conjuntos estÃ¡n vacÃ­os o son nulos.")
    }

    # Generar el diagrama de Venn
    venn_diagram <- venn.diagram(
        x = up_kegg_HvL,
        category.names = c("THP-1", "BoMac"),
        filename = NULL,
        fill = c("#fe7f2d", "#2a9d8f"),
        alpha = 0.5,
        cex = 2,
        lty = 'blank',
        # Set names
        cat.cex = 1,
        cat.fontface = "bold",
        cat.default.pos = "outer",
        cat.dist = c(-0.01, -0.01),
        scaled = TRUE
        
    )
    # Devolver solo el diagrama de Venn
    a<-grid.draw(venn_diagram)
    table <- gplots::venn(up_kegg_HvL, show.plot=FALSE)
    list(diagrama = a, tabla = table)
}
```
### Shared Gene enrichment results between cells
::::::: {.panel-tabset}
# High vs Low
:::::: {.panel-tabset}
## Up-regulated
::::: {.panel-tabset}
### KEGG
:::: {.panel-tabset}
#### Venn Diagram
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
up_HvL_KEGG<-procesar_y_generar_venn(result_upthp1HvL, result_upbomacHvL)
```

```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
#### Only THP-1
KEGG_up_thp1 <- attr(up_HvL_KEGG$tabla, "intersections")$`THP-1`
KEGG_up_thp1 <- paste0("hsa", KEGG_up_thp1)
tabla_KEGG_up_thp1 <- result_upthp1HvL@result %>% 
                  filter(ID %in% KEGG_up_thp1)
```
#### Shared THP-1 & BoMac
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
KEGG_up_shared <- attr(up_HvL_KEGG$tabla, "intersections")$`THP-1:BoMac`
KEGG_up_shared_t <- paste0("hsa", KEGG_up_shared)
KEGG_up_shared_b <- paste0("bta", KEGG_up_shared)

tabla_KEGG_up_shared_t <- result_upthp1HvL@result %>% 
                  filter(ID %in% KEGG_up_shared_t)
tabla_KEGG_up_shared_b <- result_upbomacHvL@result %>% 
                  filter(ID %in% KEGG_up_shared_b)
```
::: {.panel-tabset}
##### Table THP-1
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(tabla_KEGG_up_shared_t,"tabla_KEGG_up_shared_t")
```
##### Table BoMac
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(tabla_KEGG_up_shared_b,"tabla_KEGG_up_shared_b")
```
:::
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
#### Only BoMac
KEGG_up_bomac <- attr(up_HvL_KEGG$tabla, "intersections")$`BoMac`
KEGG_up_bomac <- paste0("bta", KEGG_up_bomac)
tabla_KEGG_up_bomac <- result_upbomacHvL@result %>% 
                  filter(ID %in% KEGG_up_bomac)
```
::::
### Biological Process
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
print("No results in BoMac")
```
### Molecular Function
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
print("No results in BoMac")
```
### Cellular Components
:::: {.panel-tabset}
#### Venn Diagram
```{r, echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
up_HvL_CC<-procesar_y_generar_venn(cc_upthp1HvL, cc_upbomacHvL)
```
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
#### Only THP-1
CC_up_thp1 <- attr(up_HvL_CC$tabla, "intersections")$`THP-1`
tabla_CC_up_thp1 <- cc_upthp1HvL@result %>% 
                  filter(ID %in% CC_up_thp1)
```
#### Shared THP-1 & BoMac
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
CC_up_shared <- attr(up_HvL_CC$tabla, "intersections")$`THP-1:BoMac`
CC_up_shared_t <- paste0("", CC_up_shared)
CC_up_shared_b <- paste0("", CC_up_shared)

tabla_CC_up_shared_t <- cc_upthp1HvL@result %>% 
                  filter(ID %in% CC_up_shared_t)
tabla_CC_up_shared_b <- cc_upbomacHvL@result %>% 
                  filter(ID %in% CC_up_shared_b)
```
::: {.panel-tabset}
##### Table THP-1
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(tabla_CC_up_shared_t,"tabla_CC_up_shared_t")
```
##### Table BoMac
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(tabla_CC_up_shared_b,"tabla_CC_up_shared_b")
```
:::

```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
#### Only BoMac
CC_up_bomac <- attr(up_HvL_CC$tabla, "intersections")$`BoMac`
tabla_CC_up_bomac <- cc_upbomacHvL@result %>% 
                  filter(ID %in% CC_up_bomac)
```
:::: 
:::::
## Down-regulated
::::: {.panel-tabset}
### KEGG
:::: {.panel-tabset}
#### Venn Diagram
```{r, message=FALSE, warning=FALSE, fig.align='center'}
down_HvL_KEGG<-procesar_y_generar_venn(result_downthp1HvL, result_downbomacHvL)
```
#### Only THP-1
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
KEGG_down_thp1 <- attr(down_HvL_KEGG$tabla, "intersections")$`THP-1`
KEGG_down_thp1 <- paste0("hsa", KEGG_down_thp1)
tabla_KEGG_down_thp1 <- result_downthp1HvL
tabla_KEGG_down_thp1@result <- tabla_KEGG_down_thp1@result %>%
                            filter(ID %in% KEGG_down_thp1)
```
::: {.panel-tabset}
##### Table
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(tabla_KEGG_down_thp1@result,"tabla_KEGG_down_thp1")
```
##### Bubbleplot
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
dotplot(tabla_KEGG_down_thp1)
```
:::
#### Shared THP-1 & BoMac
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
KEGG_down_shared <- attr(down_HvL_KEGG$tabla, "intersections")$`THP-1:BoMac`
KEGG_down_shared_t <- paste0("hsa", KEGG_down_shared)
KEGG_down_shared_b <- paste0("bta", KEGG_down_shared)

tabla_KEGG_down_shared_t <- result_downthp1HvL
tabla_KEGG_down_shared_b <- result_downbomacHvL

tabla_KEGG_down_shared_t@result <- tabla_KEGG_down_shared_t@result %>%
                                   filter(ID %in% KEGG_down_shared_t)
tabla_KEGG_down_shared_b@result <- tabla_KEGG_down_shared_b@result %>%
                                   filter(ID %in% KEGG_down_shared_b)
```
::: {.panel-tabset}
##### Table THP-1
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(tabla_KEGG_down_shared_t@result,"tabla_KEGG_down_shared_t")
```
##### Bubbleplot THP-1
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
dotplot(tabla_KEGG_down_shared_t)
```
##### Table BoMac
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(tabla_KEGG_down_shared_b@result,"tabla_KEGG_down_shared_b")
```
##### Bubbleplot BoMac
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
dotplot(tabla_KEGG_down_shared_b)
```
:::
#### Only BoMac
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
tabla_KEGG_down_bomac <- result_downbomacHvL

KEGG_down_bomac <- attr(down_HvL_KEGG$tabla, "intersections")$`BoMac`
KEGG_down_bomac <- paste0("bta", KEGG_down_bomac)

tabla_KEGG_down_bomac@result <- tabla_KEGG_down_bomac@result %>%
                                filter(ID %in% KEGG_down_bomac)
```
::: {.panel-tabset}
##### Table
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(tabla_KEGG_down_bomac@result,"tabla_KEGG_down_bomac")
```
##### Bubbleplot
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
dotplot(tabla_KEGG_down_bomac)
```
:::
::::
### Biological Process
:::: {.panel-tabset}
#### Venn Diagram
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
print("no results bomac")
#down_HvL_BP<-procesar_y_generar_venn(bp_downthp1HvL, bp_downbomacHvL)
```
::::
### Molecular Function
:::: {.panel-tabset}
#### Venn Diagram
```{r,  echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
print("no results bomac")
#down_HvL_BP<-procesar_y_generar_venn(mf_downthp1HvL, mf_downbomacHvL)
```
::::
### Cellular Components
:::: {.panel-tabset}
#### Venn Diagram
```{r,  echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
print("no results bomac")
#down_HvL_BP<-procesar_y_generar_venn(cc_downthp1HvL, cc_downbomacHvL)
```
::::
:::::
::::::
# Medium vs High
:::::: {.panel-tabset}
## Up-regulated
::::: {.panel-tabset}
### KEGG
:::: {.panel-tabset}
#### Venn diagram
```{r, message=FALSE, warning=FALSE, fig.align='center'}
up_MvH_KEGG<-procesar_y_generar_venn(result_upthp1MvH, result_upbomacMvH)
```
#### Only THP-1
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
KEGG_down_thp1 <- attr(up_MvH_KEGG$tabla, "intersections")$`THP-1`
KEGG_down_thp1 <- paste0("hsa", KEGG_down_thp1)
tabla_KEGG_down_thp1 <- result_upthp1MvH
tabla_KEGG_down_thp1@result <- tabla_KEGG_down_thp1@result %>%
                            filter(ID %in% KEGG_down_thp1)
```
::: {.panel-tabset}
##### Table
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(tabla_KEGG_down_thp1@result,"tabla_KEGG_down_thp1")
```
##### Bubbleplot
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
dotplot(tabla_KEGG_down_thp1)
```
:::
#### Shared THP-1 & BoMac
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
KEGG_down_shared <- attr(up_MvH_KEGG$tabla, "intersections")$`THP-1:BoMac`
KEGG_down_shared_t <- paste0("hsa", KEGG_down_shared)
KEGG_down_shared_b <- paste0("bta", KEGG_down_shared)

tabla_KEGG_down_shared_t <- result_upthp1MvH
tabla_KEGG_down_shared_b <- result_upbomacMvH

tabla_KEGG_down_shared_t@result <- tabla_KEGG_down_shared_t@result %>%
                                   filter(ID %in% KEGG_down_shared_t)
tabla_KEGG_down_shared_b@result <- tabla_KEGG_down_shared_b@result %>%
                                   filter(ID %in% KEGG_down_shared_b)
```
::: {.panel-tabset}
##### Table THP-1
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(tabla_KEGG_down_shared_t@result,"tabla_KEGG_down_shared_t")
```
##### Bubbleplot THP-1
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
dotplot(tabla_KEGG_down_shared_t)
```
##### Table BoMac
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(tabla_KEGG_down_shared_b@result,"tabla_KEGG_down_shared_b")
```
##### Bubbleplot BoMac
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
dotplot(tabla_KEGG_down_shared_b)
```
:::
#### Only BoMac
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
tabla_KEGG_down_bomac <- result_upbomacMvH

KEGG_down_bomac <- attr(up_MvH_KEGG$tabla, "intersections")$`BoMac`
KEGG_down_bomac <- paste0("bta", KEGG_down_bomac)

tabla_KEGG_down_bomac@result <- tabla_KEGG_down_bomac@result %>%
                                filter(ID %in% KEGG_down_bomac)
```
::: {.panel-tabset}
##### Table
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(tabla_KEGG_down_bomac@result,"tabla_KEGG_down_bomac")
```
##### Bubbleplot
```{r,echo=FALSE,message=FALSE, warning=FALSE, fig.align='center'}
dotplot(tabla_KEGG_down_bomac)
```
:::
::::
### Biological process
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
up_MvH_BP<-procesar_y_generar_venn(bp_upthp1MvH, bp_upbomacMvH)
print("No intersection")
```
### Molecular Function
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#down_MvH_MF<-procesar_y_generar_venn(mf_upthp1MvH, mf_upbomacMvH)
print("No mf_upbomacMvH")
```
### Cellular Components
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
up_MvH_CC<-procesar_y_generar_venn(cc_upthp1MvH, cc_upbomacMvH)
print("No intersection")
```
:::::
## Down-regulated
::::: {.panel-tabset}
### KEGG
```{r, message=FALSE, warning=FALSE, fig.align='center'}
#down_MvH_KEGG<-procesar_y_generar_venn(result_downthp1MvH, result_downbomacMvH)
print("No results result_downthp1MvH")
```
### Biological process
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#down_MvH_BP<-procesar_y_generar_venn(bp_downthp1MvH, bp_downbomacMvH)
print("No bp_downthp1MvH")
```
### Molecular Function
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#down_MvH_MF<-procesar_y_generar_venn(mf_downthp1MvH, mf_downbomacMvH)
print("No mf_downthp1MvH")
```
### Cellular Components
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#down_MvH_CC<-procesar_y_generar_venn(cc_downthp1MvH, cc_downbomacMvH)
print("No cc_downthp1MvH")
```
:::::
::::::
# Medium vs Low
:::::: {.panel-tabset}
## Up-regulated
::::: {.panel-tabset}
### KEGG
```{r, message=FALSE, warning=FALSE, fig.align='center'}
up_MvL_KEGG<-procesar_y_generar_venn(result_upthp1MvL, result_upbomacMvL)
```
### Biological process
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
up_MvL_BP<-procesar_y_generar_venn(bp_upthp1MvL, bp_upbomacMvL)
print("No intersection")
```
### Molecular Function
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
up_MvL_MF<-procesar_y_generar_venn(mf_upthp1MvL, mf_upbomacMvL)
print("No intersection")
```
### Cellular Components
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
up_MvL_CC<-procesar_y_generar_venn(cc_upthp1MvL, cc_upbomacMvL)
print("No intersection")
```
:::::
## Down-regulated
::::: {.panel-tabset}
### KEGG
```{r, message=FALSE, warning=FALSE, fig.align='center'}
#down_MvL_KEGG<-procesar_y_generar_venn(result_downthp1MvL, result_downbomacMvL)
print("No results downbomacMvL")
```
### Biological process
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#down_MvL_BP<-procesar_y_generar_venn(bp_downthp1MvL, bp_downbomacMvL)
print("No results downbomacMvL")
```
### Molecular Function
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
down_MvL_MF<-procesar_y_generar_venn(mf_downthp1MvL, mf_downbomacMvL)
print("No intersection")
```
### Cellular Components
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#down_MvL_CC<-procesar_y_generar_venn(cc_downthp1MvL, cc_downbomacMvL)
print("No results downbomacMvL")
```
:::::
::::::
:::::::

## Common DEGs Virulence for each Cell
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
obtener_ortologos_y_cruzar <- function(genes, dataframe) {
    # Configurar el objeto ensembl para Homo sapiens
    ensembl <- useMart("ensembl", dataset = "hsapiens_gene_ensembl")

    # Atributos para obtener los ortologos de Bos taurus
    attributes <- c('ensembl_gene_id', 'btaurus_homolog_ensembl_gene',
                    'btaurus_homolog_orthology_type')

    # Obtener ortÃ³logos de Bos taurus
    orthologs <- getBM(attributes = attributes, 
                       filters = 'ensembl_gene_id',
                       values = genes,
                       mart = ensembl)

    # Filtrar por ortologos one-to-one
    orthologs_one_to_one <- orthologs[orthologs$btaurus_homolog_orthology_type == 'ortholog_one2one', ]

    # Cruzar los dataframes
    resultado <- merge(dataframe, orthologs_one_to_one, by.x = "Row.names", by.y = "ensembl_gene_id", all = FALSE)

    # Devolver el resultado
    return(resultado)
}
cruzado_THP1_HvL <- obtener_ortologos_y_cruzar(resultados_HvL$all_data$Row.names, resultados_HvL$all_data)
cruzado_THP1_MvH <- obtener_ortologos_y_cruzar(resultados_MvH$all_data$Row.names, resultados_MvH$all_data)
cruzado_THP1_MvL <- obtener_ortologos_y_cruzar(resultados_MvL$all_data$Row.names, resultados_MvL$all_data)

asignar_colores <- function(dataframe) {
    dataframe$color2 <- ifelse(dataframe$log2FoldChange > 0 & -log10(dataframe$padj) > 1.3, "red",
                              ifelse(dataframe$log2FoldChange < 0 & -log10(dataframe$padj) > 1.3, "blue", "grey"))
    return(dataframe)
}
cruzado_THP1_HvL<-asignar_colores(cruzado_THP1_HvL)
cruzado_THP1_MvH<-asignar_colores(cruzado_THP1_MvH)
cruzado_THP1_MvL<-asignar_colores(cruzado_THP1_MvL)

bresultados_HvLn<-asignar_colores(bresultados_HvL$all_data)
bresultados_MvHn<-asignar_colores(bresultados_MvH$all_data)
bresultados_MvLn<-asignar_colores(bresultados_MvL$all_data)
```

```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#NO FOLD THRESHOLD >0, <0
up_THP1_HvL<- filter(cruzado_THP1_HvL, color2 == 'red')$btaurus_homolog_ensembl_gene
down_THP1_HvL<- filter(cruzado_THP1_HvL, color2 == 'blue')$btaurus_homolog_ensembl_gene

up_bomac_HvL<- filter(bresultados_HvLn, color2 == 'red')$Row.names
down_bomac_HvL<- filter(bresultados_HvLn, color2 == 'blue')$Row.names

up_THP1_MvH<- filter(cruzado_THP1_MvH, color2 == 'red')$btaurus_homolog_ensembl_gene
down_THP1_MvH<- filter(cruzado_THP1_MvH, color2 == 'blue')$btaurus_homolog_ensembl_gene

up_bomac_MvH<- filter(bresultados_MvHn, color2 == 'red')$Row.names
down_bomac_MvH<- filter(bresultados_MvHn, color2 == 'blue')$Row.names

up_THP1_MvL<- filter(cruzado_THP1_MvL, color2 == 'red')$btaurus_homolog_ensembl_gene
down_THP1_MvL<- filter(cruzado_THP1_MvL, color2 == 'blue')$btaurus_homolog_ensembl_gene

up_bomac_MvL<- filter(bresultados_MvLn, color2 == 'red')$Row.names
down_bomac_MvL<- filter(bresultados_MvLn, color2 == 'blue')$Row.names
```

```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
generar_venn_y_extraer_intersecciones <- function(up_THP1_MvL, down_THP1_MvL, up_bomac_MvL, down_bomac_MvL) {
    # Crear el diagrama de Venn
    venn_diagram <- venn.diagram(
        x = list(up_THP1_MvL, down_THP1_MvL, up_bomac_MvL, down_bomac_MvL),
        category.names = c("up_THP-1", "down_THP-1", "up_BoMac", "down_BoMac"),
        filename = NULL,
        fill = c("red", "blue", "#ffa8a8", "#a8abff"),
        alpha = 0.5,
        cex = 2,
        lty = 'blank',
        cat.cex = 1,
        cat.fontface = "bold",
        cat.default.pos = "outer",
        scaled = TRUE
    )
    grid.draw(venn_diagram)

    # Generar tabla de intersecciones
    table <- gplots::venn(list(
        `up_THP-1`= up_THP1_MvL,
        `down_THP-1`= down_THP1_MvL,
        `up_BoMac`= up_bomac_MvL,
        `down_BoMac`= down_bomac_MvL
    ), show.plot=FALSE)

    # Extraer intersecciones especÃ­ficas
    down_interseccion <- attr(table, "intersections")$`down_THP-1:down_BoMac`
    up_interseccion <- attr(table, "intersections")$`up_THP-1:up_BoMac`

    # Devolver las intersecciones
    return(list(down = down_interseccion, up = up_interseccion))
}
```

### Venn plots and IDs between THP-1 and BoMac
:::::: {.panel-tabset}
# Venn plots
::::: {.panel-tabset}
## High vs Low
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
res_HvL<- generar_venn_y_extraer_intersecciones(up_THP1_HvL,down_THP1_HvL,up_bomac_HvL,down_bomac_HvL)
```
## Medium vs High
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
res_MvH<- generar_venn_y_extraer_intersecciones(up_THP1_MvH,down_THP1_MvH,up_bomac_MvH,down_bomac_MvH)
```
## Medium vs Low
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
res_MvL<- generar_venn_y_extraer_intersecciones(up_THP1_MvL,down_THP1_MvL,up_bomac_MvL,down_bomac_MvL)
```
::::::
# IDs List
::::: {.panel-tabset}
## High vs Low
:::: {.panel-tabset}
### Up-regulated
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
table_HvL_up <- get_one_to_one_orthologs(res_HvL$up)
create_datatable(table_HvL_up,"table_HvL_up")
```
### Down-regulated
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
table_HvL_down <- get_one_to_one_orthologs(res_HvL$down)
create_datatable(table_HvL_down,"table_HvL_down")
```
::::
## Medium vs High
:::: {.panel-tabset}
### Up-regulated
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
table_MvH_up <- get_one_to_one_orthologs(res_MvH$up)
create_datatable(table_MvH_up,"table_MvH_up")
```
### Down-regulated
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
table_MvH_down <- get_one_to_one_orthologs(res_MvH$down)
create_datatable(table_MvH_down,"table_MvH_down")
```
::::
## Medium vs Low
:::: {.panel-tabset}
### Up-regulated
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
table_MvL_up <- get_one_to_one_orthologs(res_MvL$up)
create_datatable(table_MvL_up,"table_MvL_up")
```
### Down-regulated
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
table_MvL_down <- get_one_to_one_orthologs(res_MvL$down)
create_datatable(table_MvL_down,"table_MvL_down")
```
::::
:::::
:::::::
### Venn plots between Virulence
::::::: {.panel-tabset}
# Up-regulated
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
venn_diagram <- venn.diagram(
        x = list(res_HvL$up,res_MvH$up,res_MvL$up),
        category.names = c("up_HvL", "up_MvH","up_MvL"),
        filename = NULL,
        fill = c("#a8abff","orange","yellow"),
        alpha = 0.5,
        cex = 2,
        lty = 'blank',
        # Set names
        cat.cex = 1,
        cat.fontface = "bold",
        cat.default.pos = "outer",
        scaled = TRUE
        
    )
grid.draw(venn_diagram)
```
# Down-regulated
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
venn_diagram <- venn.diagram(
        x = list(res_HvL$down,res_MvL$down),
        category.names = c("down_HvL","down_MvL"),
        filename = NULL,
        fill = c("orange","yellow"),
        alpha = 0.5,
        cex = 2,
        lty = 'blank',
        scaled = TRUE
        
    )
grid.draw(venn_diagram)
```
:::::::


### Gene enrichment for shared DEGs between cells
::::::: {.panel-tabset}
# High vs Low
:::::: {.panel-tabset}
## Up-regulated
::::: {.panel-tabset}
### KEGG
:::: {.panel-tabset}
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
up_res_HvL<-get_one_to_one_orthologs(res_HvL$up)
kegg_up_res_HvL<-enrich_kegg(up_res_HvL$hsapiens_homolog_ensembl_gene, org.Hs.eg.db, 'hsa')
```
#### Table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(kegg_up_res_HvL@result,"kegg_up_res_HvL")
```
#### Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
dotplot(kegg_up_res_HvL)
```
::::
### Biological Process
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
cc_up_res_HvL<-enrich_GO_and_convert(up_res_HvL$hsapiens_homolog_ensembl_gene,org.Hs.eg.db,"BP")
print("No results")
```
### Molecular Function
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
cc_up_res_HvL<-enrich_GO_and_convert(up_res_HvL$hsapiens_homolog_ensembl_gene,org.Hs.eg.db,"MF")
```
:::: {.panel-tabset}
#### Table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(cc_up_res_HvL@result,"mf_up_res_HvL")
```

#### Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
dotplot(cc_up_res_HvL)
```
#### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
rrBo_up_BP_thp1_HvL<-perform_analysis(cc_up_res_HvL@result, "org.Hs.eg.db", "MF")
wordcloudPlot(rrBo_up_BP_thp1_HvL$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
```
#### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
scatterPlot(rrBo_up_BP_thp1_HvL$simMatrix, rrBo_up_BP_thp1_HvL$reducedTerms)
```
#### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
treemapPlot(rrBo_up_BP_thp1_HvL$reducedTerms)
```
::::
### Cellular Components
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
cc_up_res_HvL<-enrich_GO_and_convert(up_res_HvL$hsapiens_homolog_ensembl_gene,org.Hs.eg.db,"CC")
```
:::: {.panel-tabset}
#### Table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(cc_up_res_HvL@result,"cc_up_res_HvL")
```
#### Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
dotplot(cc_up_res_HvL)
```
#### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
rrBo_up_BP_thp1_HvL<-perform_analysis(cc_up_res_HvL@result, "org.Hs.eg.db", "CC")
wordcloudPlot(rrBo_up_BP_thp1_HvL$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
```
#### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
scatterPlot(rrBo_up_BP_thp1_HvL$simMatrix, rrBo_up_BP_thp1_HvL$reducedTerms)
```
#### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
treemapPlot(rrBo_up_BP_thp1_HvL$reducedTerms)
```
::::
:::::
## Down-regulated
::::: {.panel-tabset}
### KEGG
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
up_res_HvL<-get_one_to_one_orthologs(res_HvL$down)
#kegg_up_res_HvL<-enrich_kegg(up_res_HvL$hsapiens_homolog_ensembl_gene, org.Hs.eg.db, 'hsa')
print("No results")
```
### Biological Process
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#cc_up_res_HvL<-enrich_GO_and_convert(up_res_HvL$hsapiens_homolog_ensembl_gene,org.Hs.eg.db,"BP")
print("No results")
```
### Molecular Function
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
#cc_up_res_HvL<-enrich_GO_and_convert(up_res_HvL$hsapiens_homolog_ensembl_gene,org.Hs.eg.db,"MF")
print("No results")
```
### Cellular Components
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
cc_up_res_HvL<-enrich_GO_and_convert(up_res_HvL$hsapiens_homolog_ensembl_gene,org.Hs.eg.db,"CC")
```
:::: {.panel-tabset}
#### Table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(cc_up_res_HvL@result,"cc_down_res_HvL")
```
#### Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
dotplot(cc_up_res_HvL)
```
#### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
rrBo_up_BP_thp1_HvL<-perform_analysis(cc_up_res_HvL@result, "org.Hs.eg.db", "CC")
wordcloudPlot(rrBo_up_BP_thp1_HvL$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
```
#### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
scatterPlot(rrBo_up_BP_thp1_HvL$simMatrix, rrBo_up_BP_thp1_HvL$reducedTerms)
```
#### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
treemapPlot(rrBo_up_BP_thp1_HvL$reducedTerms)
```
::::
:::::
::::::
# Medium vs High
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
print("no results")
```
# Medium vs Low
:::::: {.panel-tabset}
## Up-regulated
::::: {.panel-tabset}
### KEGG
:::: {.panel-tabset}
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
up_res_MvL<-get_one_to_one_orthologs(res_MvL$up)
kegg_up_res_MvL<-enrich_kegg(up_res_MvL$hsapiens_homolog_ensembl_gene, org.Hs.eg.db, 'hsa')
```
#### Table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(kegg_up_res_MvL@result,"kegg_up_res_MvL")
```
#### Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
dotplot(kegg_up_res_MvL)
```
::::
### Biological Process
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
cc_up_res_MvL<-enrich_GO_and_convert(up_res_MvL$hsapiens_homolog_ensembl_gene,org.Hs.eg.db,"BP")
```
:::: {.panel-tabset}
#### Table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(cc_up_res_MvL@result,"bp_up_res_MvL")
```
#### Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
dotplot(cc_up_res_MvL)
```
::::
### Molecular Function
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
cc_up_res_MvL<-enrich_GO_and_convert(up_res_MvL$hsapiens_homolog_ensembl_gene,org.Hs.eg.db,"MF")
```
### Cellular Components
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
cc_up_res_MvL<-enrich_GO_and_convert(up_res_MvL$hsapiens_homolog_ensembl_gene,org.Hs.eg.db,"CC")
```
:::: {.panel-tabset}
#### Table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(cc_up_res_MvL@result,"cc_up_res_MvL")
```
#### Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
dotplot(cc_up_res_MvL)
```
#### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
rrBo_up_BP_thp1_MvL<-perform_analysis(cc_up_res_MvL@result, "org.Hs.eg.db", "CC")
wordcloudPlot(rrBo_up_BP_thp1_MvL$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
```
#### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
scatterPlot(rrBo_up_BP_thp1_MvL$simMatrix, rrBo_up_BP_thp1_MvL$reducedTerms)
```
#### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
treemapPlot(rrBo_up_BP_thp1_MvL$reducedTerms)
```
::::
:::::
## Down-regulated
::::: {.panel-tabset}
### KEGG
:::: {.panel-tabset}
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
up_res_MvL<-get_one_to_one_orthologs(res_MvL$down)
kegg_up_res_MvL<-enrich_kegg(up_res_MvL$hsapiens_homolog_ensembl_gene, org.Hs.eg.db, 'hsa')
```
#### Table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(kegg_up_res_MvL@result,"kegg_down_res_MvL")
```
#### Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
dotplot(kegg_up_res_MvL)
```
::::
### Biological Process
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
cc_up_res_MvL<-enrich_GO_and_convert(up_res_MvL$hsapiens_homolog_ensembl_gene,org.Hs.eg.db,"BP")
```
:::: {.panel-tabset}
#### Table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(cc_up_res_MvL@result,"bp_down_res_MvL")
```
#### Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
dotplot(cc_up_res_MvL)
```
::::
### Molecular Function
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
cc_up_res_MvL<-enrich_GO_and_convert(up_res_MvL$hsapiens_homolog_ensembl_gene,org.Hs.eg.db,"MF")
```
:::: {.panel-tabset}
#### Table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(cc_up_res_MvL@result,"mf_down_res_MvL")
```
#### Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
dotplot(cc_up_res_MvL)
```
::::
### Cellular Components
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
cc_up_res_MvL<-enrich_GO_and_convert(up_res_MvL$hsapiens_homolog_ensembl_gene,org.Hs.eg.db,"CC")
```
:::: {.panel-tabset}
#### Table
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
create_datatable(cc_up_res_MvL@result,"cc_down_res_MvL")
```
#### Bubble Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
dotplot(cc_up_res_MvL)
```
#### Word Cloud
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
rrBo_up_BP_thp1_MvL<-perform_analysis(cc_up_res_MvL@result, "org.Hs.eg.db", "CC")
wordcloudPlot(rrBo_up_BP_thp1_MvL$reducedTerms,scale=c(3,.4), min.freq=1,use.r.layout=T)
```
#### Scatter Plot
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
scatterPlot(rrBo_up_BP_thp1_MvL$simMatrix, rrBo_up_BP_thp1_MvL$reducedTerms)
```
#### Tree Map
```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.align='center'}
treemapPlot(rrBo_up_BP_thp1_MvL$reducedTerms)
```
::::
:::::
::::::
:::::::
## Packages used in R
```{r}
session_info()
```
